### PR TITLE
Added a new variable for the max memory and fixed get_memory_limit fu…

### LINKED ISSUE #160: Add another api param for memory size limit for enterprise so we can use custom instances

### PR DESCRIPTION
**Pull request by MuhammadQadora**
fix #160 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new environment variable `MEMORY_LIMIT` for better memory management.
	- Added a `maxMemory` parameter in the configuration for enhanced control over memory allocation in FalkorDB deployments.

- **Bug Fixes**
	- Improved logic for determining memory limits based on instance types, ensuring defaults are applied correctly.

- **Documentation**
	- Updated configuration documentation to include new `maxMemory` parameter details.

### CODE REVIEW

File: falkordb-node/node-entrypoint.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ FALKORDB_RESULT_SET_SIZE=${FALKORDB_RESULT_SET_SIZE:-10000}
 FALKORDB_QUERY_MEM_CAPACITY=${FALKORDB_QUERY_MEM_CAPACITY:-0}
 FALKORDB_TIMEOUT_MAX=${FALKORDB_TIMEOUT_MAX:-0}
 FALKORDB_TIMEOUT_DEFAULT=${FALKORDB_TIMEOUT_DEFAULT:-0}
-
+E_MEMORY_LIMIT=${E_MEMORY_LIMIT:-''}
```
Comment (dudizimber) [inline falkordb-node/node-entrypoint.sh:41]: What's "E_"?

File: omnistrate.enterprise.yaml [hunk 2]
```diff
@@ -275,6 +275,14 @@ x-clusterparam: &clusterparam
...
       min: 0
       max: 2000000000
+    - key: maxMemory
+      description: Choose the maxmemory for your database (do not exceed InstanceType memory)
+      name: FalkorDB Maxmemory
+      type: String
+      modifiable: false
+      required: false
+      export: true
+      defaultValue: "0GB"
```
Comment (dudizimber) [inline omnistrate.enterprise.yaml:285]: Let's set the default to the min we're usually going to deploy which is 8gb - 2gb = 6gb

Comment (MuhammadQadora) [reply omnistrate.enterprise.yaml:285]: Ok

File: falkordb-node/node-entrypoint.sh [hunk 3]
```diff
@@ -215,7 +216,9 @@ get_memory_limit() {
   )
 
   MEMORY_LIMIT=${memory_limit_instance_type_map[$INSTANCE_TYPE]}
-  if [[ -z $MEMORY_LIMIT ]]; then
+  if [[ -n "$E_MEMORY_LIMIT" ]]; then
+    MEMORY_LIMIT=$E_MEMORY_LIMIT
```
Comment (dudizimber) [inline falkordb-node/node-entrypoint.sh:220]: Needs to be added to the cluster_endpoint as well

**Review by dudizimber — APPROVED** [review]

**Comment by MuhammadQadora** [comment]
@dudizimber Waiting for the nodegroup issue to be solved.

**Pull request merged**